doc/source/release/1.15.0-notes.rst

```diff
@@ -17,96 +17,1431 @@ run your code with ``python -Wd`` and check for ``DeprecationWarning`` s).
+threads
+(see the ``threading`` stdlib module). This support was enabled by fixing a
+significant number of thread-safety issues in both pure Python and
+C/C++/Cython/Fortran extension modules. Wheels are provided on PyPI for this
+release; NumPy ``>=2.1.3`` is required at runtime. Note that building for a
+free-threaded interpreter requires a recent pre-release or nightly for Cython
+``3.1.0``.
+
+Support for free-threaded Python does not mean that SciPy is fully thread-safe.
+Please see :ref:`scipy_thread_safety` for more details.
 
+If you are interested in free-threaded Python, for example because you have a
+multiprocessing-based workflow that you are interested in running with Python
+threads, we encourage testing and experimentation. If you run into problems
+that you suspect are because of SciPy, please open an issue, checking first if
+the bug also occurs in the "regular" non-free-threaded CPython ``3.13`` build.
+Many threading bugs can also occur in code that releases the GIL; disabling
+the GIL only makes it easier to hit threading bugs.
 
+**************************
+Array API Standard Support
+**************************
+Experimental support for array libraries other than NumPy has been added to
+existing sub-packages in recent versions of SciPy. Please consider testing
+these features by setting an environment variable ``SCIPY_ARRAY_API=1`` and
+providing PyTorch, JAX, or CuPy arrays as array arguments. Features with
+support added for SciPy ``1.15.0`` include:
+
+- All functions in `scipy.differentiate` (new sub-package)
+- All functions in `scipy.optimize.elementwise` (new namespace)
+- `scipy.optimize.rosen`, `scipy.optimize.rosen_der`, and
+  `scipy.optimize.rosen_hess`
+- `scipy.special.logsumexp`
+- `scipy.integrate.trapezoid`
+- `scipy.integrate.tanhsinh` (newly public function)
+- `scipy.integrate.cubature` (new function)
+- `scipy.integrate.nsum` (new function)
+- `scipy.special.chdtr`, `scipy.special.betainc`, and `scipy.special.betaincc`
+- `scipy.stats.boxcox_llf`
+- `scipy.stats.differential_entropy`
+- `scipy.stats.zmap`, `scipy.stats.zscore`, and `scipy.stats.gzscore`
+- `scipy.stats.tmean`, `scipy.stats.tvar`, `scipy.stats.tstd`,
+  `scipy.stats.tsem`, `scipy.stats.tmin`, and `scipy.stats.tmax`
+- `scipy.stats.gmean`, `scipy.stats.hmean` and `scipy.stats.pmean`
+- `scipy.stats.combine_pvalues`
+- `scipy.stats.ttest_ind`, `scipy.stats.ttest_rel`
+- `scipy.stats.directional_stats`
+- `scipy.ndimage` functions will now delegate to ``cupyx.scipy.ndimage``,
+  and for other backends will transit via NumPy arrays on the host.
+
+
+*******************
+Deprecated features
+*******************
+- Functions `scipy.linalg.interpolative.rand` and
+  `scipy.linalg.interpolative.seed` have been deprecated and will be removed
+  in SciPy ``1.17.0``.
+- Complex inputs to `scipy.spatial.distance.cosine` and
+  `scipy.spatial.distance.correlation` have been deprecated and will raise
+  an error in SciPy ``1.17.0``.
```

Comment:
I believe that `distance.kulczynski1` and `distance.sokalmichener` were deprecated in #21572

Reply:
this can be resolved now (I have no super cow powers here)